app/actions/handlers.py

```diff
@@ -0,0 +1,138 @@
+                        upper_date = min(present_time, lower_date + timedelta(days=7))
+                        positions = await client.get_positions(device.nDeviceID, auth, integration, lower_date, upper_date, True)
+                        logger.info(f"Extracted {len(positions)} obs from Lotek for device: {device.nDeviceID} between {lower_date} and {upper_date}.")
+                        for position in positions:
+                            cdip_pos = transform(position, integration)
+                            if cdip_pos:
+                                cdip_positions.append(cdip_pos)
+                        logger.debug(f"Extracted {len(cdip_positions)} of {len(positions)} points between {lower_date} and {upper_date}.")
+                        lower_date = upper_date
+
+                    if cdip_positions:
+                        logger.info(f"Observations pulled successfully for integration ID: {integration.id}.")
+
+                        def generate_batches(iterable, n=action_config.observations_per_request):
+                            for i in range(0, len(iterable), n):
+                                yield iterable[i: i + n]
+
+                        for i, batch in enumerate(generate_batches(cdip_positions)):
+                            async for attempt in stamina.retry_context(on=httpx.HTTPError, attempts=3, wait_initial=timedelta(seconds=10), wait_max=timedelta(seconds=10)):
+                                with attempt:
+                                    try:
+                                        logger.info(f'Sending observations batch #{i}: {len(batch)} observations. Device: {device.nDeviceID}')
+                                        await gundi_tools.send_observations_to_gundi(observations=batch, integration_id=integration.id)
+                                    except httpx.HTTPError as e:
+                                        msg = f'Sensors API returned error for integration_id: {integration.id}. Exception: {e}'
+                                        logger.exception(msg, extra={
+                                            'needs_attention': True,
+                                            'integration_id': integration.id,
+                                            'action_id': "pull_observations"
+                                        })
+                                        raise e
+                                    else:
+                                        observations_extracted += len(cdip_positions)
+                    else:
+                        logger.info(f"No positions fetched for device {device.nDeviceID} integration ID: {integration.id}.")
+                await state_manager.set_state(
+                    str(integration.id),
+                    "pull_observations",
+                    upper_date
+                )
+                return {'observations_extracted': observations_extracted}
+    except httpx.HTTPError as e:
```

Comment:
This try/except block seems too big and it can make it difficult to find what triggered the exception. I'd suggest wrapping smaller portions of the code with try/except blocks if needed. For example, wrap a single line that calls lotek client to get data, and add proper error handling or retries as needed. Also, check the exception type you are catching makes sense, and ensure you are catching it for a reason (for example to retry or to log something useful in gcp or in the activity logs), else it may be better to let it raise.